### PR TITLE
Reworking TEG Circulators to behave like it has passive gate and pump

### PR DESCRIPTION
**Pull request by Azrun**
[balance][bug][rework][input wanted]

## About the PR 
TEG Circulators will behave as though it contains a passive gate and a pump.  When passive gate can not be overcome due to insufficient pressure a pump will be used.  Pump will draw power from equipment power pool when active.

No air transfer will occur if passive gate is not overcome. Thus engine will not produce power.
Pump will be disabled with loss of power (APC Failure, Insufficient power in pool).
Passive gates on the hot and cold loops into engine room will be more important.

Engineers will have to overcome pressure differential with:
- Sufficient power to run motor.
- Sufficient amount of gas/heat added to inlet side of the system.
- Sufficient heat/gas removed from outlet side. (Vent gas into space to get that suction going!)

## Why's this needed? 
Engine should not take credit for negative pressure delta on cold loop AND benefit from temperate on the inlet.

## Changelog

```
(u)Azrun:
(*)Rework TEG circulator behavior to reduce "magic".  Please ensure circulators have power and/or there is less pressure on the outlet side of the circulator to make the engine happy.
```

### CODE REVIEW

File: code/WorkInProgress/TempEngine.dm [hunk 1]
```diff
@@ -13,6 +13,8 @@
 
 	var/side = null // 1=left 2=right
 	var/last_pressure_delta = 0
+	var/fan_efficiency = 10 // 0.9 ideal, but I don't want everyone to suffer... yet.
```
Comment (ZeWaka) [inline code/WorkInProgress/TempEngine.dm:16]: 0.9 or 9?

**Review by ZeWaka — APPROVED** [review]
Code looks good, and neat expansion of gameplay.

File: code/WorkInProgress/TempEngine.dm (hunk 1 above)
Comment (Azrun) [reply code/WorkInProgress/TempEngine.dm:16]: 0.9 is efficient for a fan with perfect being perfect FAN would be 1
This is a super duper fan but was a compromise to scale based pump.dm to not punish players until behavior was better understood.

**Comment by UrsulaMejor** [comment]
Have you tested this with standard furnace setups?

**Comment by Azrun** [comment]
@UrsulaMejor 
Only with respect to logic  flow, so no.  Typical use cases should only impact cold-loop behavior.

You curious about effective power outputs with possible power consumption?

**Comment by Azrun** [comment]
image
Only opened valves (hot loop bypass burn chamber, and cold loop to space) and loaded one box of char.  Was around the 8 minute mark.

**Comment by UrsulaMejor** [comment]
looks fine to me

**Comment by ZeWaka** [comment]
LGTM, going to let it sit a day more for potential feedback/etc.

**Pull request merged**